File: client-clojure/src/sirqul_io_t_platform/specs/note_full_response.clj
(ns sirqul-io-t-platform.specs.note-full-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.name-string-value-response :refer :all]
            [sirqul-io-t-platform.specs.account-short-response :refer :all]
            [sirqul-io-t-platform.specs.asset-short-response :refer :all]
            [sirqul-io-t-platform.specs.like-response :refer :all]
            )
  (:import (java.io File)))


(def note-full-response-data
  {
   (ds/opt :valid) boolean?
   (ds/opt :message) string?
   (ds/opt :version) float?
   (ds/opt :serializeNulls) boolean?
   (ds/opt :startTimeLog) int?
   (ds/opt :errorCode) string?
   (ds/opt :request) (s/coll-of name-string-value-response-spec)
   (ds/opt :noteId) int?
   (ds/opt :comment) string?
   (ds/opt :tags) string?
   (ds/opt :likesCount) int?
   (ds/opt :dislikesCount) int?
   (ds/opt :dateCreated) int?
   (ds/opt :createdDate) int?
   (ds/opt :updatedDate) int?
   (ds/opt :owner) account-short-response-spec
   (ds/opt :latitude) float?
   (ds/opt :longitude) float?
   (ds/opt :locationDescription) string?
   (ds/opt :attachedAssets) (s/coll-of asset-short-response-spec)
   (ds/opt :likes) (s/coll-of like-response-spec)
   (ds/opt :hasLiked) boolean?
   (ds/opt :hasDisliked) boolean?
   (ds/opt :returning) string?
   })

(def note-full-response-spec
  (ds/spec
    {:name ::note-full-response
     :spec note-full-response-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/NoteFullResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import NoteFullResponse._

case class NoteFullResponse (
  valid: Option[Boolean],
message: Option[String],
version: Option[Double],
serializeNulls: Option[Boolean],
startTimeLog: Option[Long],
errorCode: Option[String],
request: Option[List[NameStringValueResponse]],
noteId: Option[Long],
comment: Option[String],
tags: Option[String],
likesCount: Option[Long],
dislikesCount: Option[Long],
dateCreated: Option[Long],
createdDate: Option[Long],
updatedDate: Option[Long],
owner: Option[AccountShortResponse],
latitude: Option[Double],
longitude: Option[Double],
locationDescription: Option[String],
attachedAssets: Option[List[AssetShortResponse]],
likes: Option[List[LikeResponse]],
hasLiked: Option[Boolean],
hasDisliked: Option[Boolean],
returning: Option[String])

object NoteFullResponse {
  import DateTimeCodecs._

  implicit val NoteFullResponseCodecJson: CodecJson[NoteFullResponse] = CodecJson.derive[NoteFullResponse]
  implicit val NoteFullResponseDecoder: EntityDecoder[NoteFullResponse] = jsonOf[NoteFullResponse]
  implicit val NoteFullResponseEncoder: EntityEncoder[NoteFullResponse] = jsonEncoderOf[NoteFullResponse]
}
